src/index.js

```diff
@@ -46,25 +46,32 @@ function runSync (command, options) {
 }
 
 function runAsync (command, options) {
-  // Prepare options for exec command (don't need async and stdio as it doesn't handle them)
-  const execOptions = Object.assign({}, options)
-  delete execOptions.async
-  delete execOptions.stdio
-
   return new Promise((resolve, reject) => {
-    const asyncProcess = childProcess.exec(command, execOptions, (error, stdout) => {
-      if (error) {
-        reject(error)
-      } else {
-        resolve(stdout.toString())
+    const spawnOptions = Object.assign({shell: true}, options)
+    const timeout = spawnOptions.timeout
+    delete spawnOptions.async
+
+    return new Promise((resolve, reject) => {
+      const asyncProcess = childProcess.spawn(command, spawnOptions)
+      asyncProcess.on('error', (error) => {
+        reject(new Error(`Failed to start command: ${command}; ${error}`))
+      })
+
+      asyncProcess.on('close', (exitCode) => {
+        if (exitCode === 0) {
+          resolve(exitCode)
```

Comment:
To resolve that in a elegant manner we could create `LeakyBuffer` class in `utils` which could act as data structure with the `push` method. For example:

```javascript
const buffer = new LeakyBuffer(MAX_BUFFER)
buffer.push('some string')
buffer.toString() // -> return all gathered data
```